transcom/mymove
# MB-16296-service-area-dest-uses-final-destination

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16296)

## Summary
This PR updates the calculations for the ServiceAreaDest param so that it is utilizing the correct inputs for pricing a DDDSIT service item.

The param ServiceAreaDest stores the service area associated with the destination address of the shipment.DDDSIT service items also need the service area but using the sit_destination_final_address instead of the shipment’s destination address. A new service item param was created that utilizes ServiceAreaDest as part of this PR.


For general information reference: [MB-15990 Discovery | How should we test this?](https://dp3.atlassian.net/wiki/spaces/TH/pages/2149777409/MB-15990+Discovery#How-should-we-test-this%3F)


## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing) - Note: When running the playwright tests make sure not to use the command `make e2e test` if you are also testing in the UI, otherwise the sample moves  you created will get deleted. Backend tests will generally run with a command like this: `go test ./pkg/services/ghcrateengine --testify.m [NAME OF TEST] -v`

### How to test

1. Create a payment request with a DDDSIT service item.
2. Log in as the Service Counselor to review the move -> log in as the Prime to review the move and add the weights --> Log in as the TOO to approve the service items --> Log in as the Prime to create the payment request
3.Once the payment request is created, log in as a TIO,  go to the network tab when reviewing the payment request that was created, and make sure that you see the DDDSIT Service item and that there is a Destination SIT schedule.
4. For thorough testing if you make 2 moves with the same starting zip and different ending zips, you should see 2 different Destination SIT schedule when you're the TIO.


### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
